benchmarkapps/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs

```diff
@@ -50,64 +50,74 @@ public partial class BenchmarkApplication : IHttpConnection
                 }
 
                 var result = await task;
-                if (!ParseHttpRequest(ref result))
+                var buffer = result.Buffer;
+                while (true)
                 {
-                    break;
-                }
+                    if (!ParseHttpRequest(ref buffer, result.IsCompleted, out var examined))
+                    {
+                        return;
+                    }
 
-                if (_state == State.Body)
-                {
-                    await ProcessRequestAsync();
+                    if (_state == State.Body)
+                    {
+                        await ProcessRequestAsync();
+
+                        _state = State.StartLine;
+
+                        if (!buffer.IsEmpty)
+                        {
+                            // More input data to parse
+                            continue;
+                        }
+                    }
 
-                    _state = State.StartLine;
+                    // No more input or incomplete data, Advance the Reader
+                    Reader.AdvanceTo(buffer.Start, examined);
+                    break;
                 }
             }
         }
 
-        // Should be `in` but ReadResult isn't readonly struct
-        private bool ParseHttpRequest(ref ReadResult result)
+        private bool ParseHttpRequest(ref ReadOnlySequence<byte> buffer, bool isCompleted, out SequencePosition examined)
         {
-            var buffer = result.Buffer;
+            examined = buffer.End;
+
             var consumed = buffer.Start;
-            var examined = buffer.End;
             var state = _state;
 
             if (!buffer.IsEmpty)
             {
-                var parsingStartLine = state == State.StartLine;
-                if (parsingStartLine)
+                if (state == State.StartLine)
                 {
                     if (Parser.ParseRequestLine(new ParsingAdapter(this), buffer, out consumed, out examined))
                     {
                         state = State.Headers;
                     }
+
+                    buffer = buffer.Slice(consumed);
```

Comment:
Then it would be a different pattern to ParseHeaders? 

Also `consumed` is the value for `buffer.Start` which it is then exchanged for in this PR. e.g.
```
Reader.AdvanceTo(consumed, examined);
```
becomes
```
Reader.AdvanceTo(buffer.Start, examined);
```
So its for maintaining the `AdvanceTo` behaviour